pulsar/client.go

```diff
@@ -74,9 +74,9 @@ func NewAuthenticationAthenz(authParams map[string]string) Authentication {
 }
 
 // NewAuthenticationOAuth2 Creates OAuth2 Authentication provider
-func NewAuthenticationOAuth2(authParams map[string]string) Authentication {
-	oauth, _ := auth.NewAuthenticationOAuth2WithParams(authParams)
-	return oauth
+func NewAuthenticationOAuth2(authParams map[string]string) (Authentication, error) {
+	oauth, err := auth.NewAuthenticationOAuth2WithParams(authParams)
```

Comment:
I don't think you should change the return value.

You can add a new method or use `panic` here.

Reply:
Okay. I dont think I will do a panic/fatal or in any other way exit with an error in a function like that. But I guess that could be a matter of personal preference or coding style.

So just to be clear we're talking about the same; your suggestion would be to make another function called something like NewAuthenticationOAuth2withErrorHandling, correct?

Reply:
In principal, the modifications that modify the public APIs should not be accepted. /cc @shibd